client/game_logic.py

```diff
@@ -0,0 +1,190 @@
+import threading
+import time
+import math
+import random
+from itertools import accumulate
+
+MAX_SCORE = 100
+
+
+class Game(threading.Thread):
+    def __init__(self, pressed_keys: list, pressed_keys_lock: threading.Lock):
+        super().__init__()
+        self.pressed_key = pressed_keys
+        self.pressed_keys_lock = pressed_keys_lock
+        self.package_number = 0
+        self.score = 0
+        self.combo = 1
+        self.number_of_moles = 9
+        self.values_changed = False
+        self.mole_life_time = [2, 3]
+        self.max_moles_amount = 3
+        self.last_print_time = 0
+        self.moles_status = [[0, [0, 0, 0]]
+                             for _ in range(self.number_of_moles)]  # [status, alive_time]
```

Comment:
What is alive_time? Isn't it just a number dictating when a mole hides?
How many states moles have?

Reply:
moles hold 4 status actually, 3 is straight when they appear, after 1/3 of live time goes out it should change to 2, after 2/3 to 1 after whole time to 0.